code/controllers/subsystem/sounds.dm

```diff
@@ -0,0 +1,165 @@
+#define DATUMLESS "NO_DATUM"
+
+SUBSYSTEM_DEF(sounds)
+	name = "Sounds"
+	init_order = INIT_ORDER_SOUNDS
+	flags = SS_NO_FIRE
+	offline_implications = "Sounds may not play correctly. Shuttle call recommended."
+
+	var/using_channels_max = CHANNEL_HIGHEST_AVAILABLE // BYOND max channels
+	/// Amount of channels to reserve for random usage rather than reservations being allowed to reserve all channels. Also a nice safeguard for when someone screws up.
+	var/random_channels_min = 50
+	// Hey uh these two needs to be initialized fast because the whole "things get deleted before init" thing.
+	/// Assoc list, "[channel]" = either the datum using it or TRUE for an unsafe-reserved (datumless reservation) channel
+	var/list/using_channels
+	/// Assoc list datum = list(channel1, channel2, ...) for what channels something reserved.
+	var/list/using_channels_by_datum
+	// Special datastructure for fast channel management
+	/// List of all channels as numbers
+	var/list/channel_list
+	/// Associative list of all reserved channels associated to their position. "[channel_number]" = index as number
+	var/list/reserved_channels
+	/// lower iteration position - Incremented and looped to get "random" sound channels for normal sounds. The channel at this index is returned when asking for a random channel.
+	var/channel_random_low
+	/// higher reserve position - decremented and incremented to reserve sound channels, anything above this is reserved. The channel at this index is the highest unreserved channel.
+	var/channel_reserve_high
+
+/datum/controller/subsystem/sounds/Initialize()
+	setup_available_channels()
+	return ..()
+
+/**
+  * Sets up all available sound channels
+  */
+/datum/controller/subsystem/sounds/proc/setup_available_channels()
+	channel_list = list()
+	reserved_channels = list()
+	using_channels = list()
+	using_channels_by_datum = list()
```

Comment:
Initialize these in the definition, or use a lazy list here

Reply:
Not sure it's worth it? Sounds are played basically immediately so it would barely make a difference if these lists were lazily initialized. I don't see any harm in leaving these here

Reply:
Having this in a separate proc from `Initialize` had me under the impression that this was called in multiple places, hence me requesting it be lazy-ified. This is only called in `Initialize`, so I question why you made this into an extra proc.